lib/logstash/inputs/sqs.rb

```diff
@@ -76,101 +77,108 @@ class LogStash::Inputs::SQS < LogStash::Inputs::Threadable
   # Name of the event field in which to store the  SQS message Sent Timestamp
   config :sent_timestamp_field, :validate => :string
 
-  public
-  def aws_service_endpoint(region)
-    return {
-        :sqs_endpoint => "sqs.#{region}.amazonaws.com"
-    }
-  end
+  MAX_TIME_BEFORE_GIVING_UP = 60
+  MAX_MESSAGES_TO_FETCH = 10 # Between 1-10 in the AWS-SDK doc
+  BACKOFF_SLEEP_TIME = 1
+  BACKOFF_FACTOR = 2
 
-  public
   def register
-    @logger.info("Registering SQS input", :queue => @queue)
     require "aws-sdk"
+    @logger.info("Registering SQS input", :queue => @queue)
 
-    @sqs = AWS::SQS.new(aws_options_hash)
+    verify_credentials!
+  end
 
-    begin
-      @logger.debug("Connecting to AWS SQS queue", :queue => @queue)
-      @sqs_queue = @sqs.queues.named(@queue)
-      @logger.info("Connected to AWS SQS queue successfully.", :queue => @queue)
-    rescue Exception => e
-      @logger.error("Unable to access SQS queue.", :error => e.to_s, :queue => @queue)
-      throw e
-    end # begin/rescue
-  end # def register
+  def aws_sqs_client
+    @sqs ||= Aws::SQS::Client.new(aws_options_hash)
+  end
 
-  public
-  def run(output_queue)
-    @logger.debug("Polling SQS queue", :queue => @queue)
+  def queue_url
+    @queue_url ||= begin
+                     aws_sqs_client.get_queue_url(:queue_name =>  @queue)[:queue_url]
+                   rescue Aws::SQS::Errors::ServiceError => e
+                     @logger.error("Cannot establish connection to Amazon SQS", :error => e)
+                     raise LogStash::ConfigurationError, "Verify the SQS queue name and your credentials"
+                   end
+  end
+
+  def verify_credentials!
```

Comment:
I find it strange that the `verify_credentials!` has the side effect of populating `@queue_url`.

maybe make register do a call to a `setup_queue` method that, as a consequence, can report a credential failure?